spec/services/discovery_engine/autocomplete/complete_spec.rb

```diff
@@ -0,0 +1,40 @@
+RSpec.describe DiscoveryEngine::Autocomplete::Complete do
+  subject(:completion) { described_class.new(query, client:) }
+
+  let(:client) { double("CompletionService::Client", complete_query:) }
+
+  before do
+    allow(Rails.configuration).to receive(:discovery_engine_datastore).and_return("/the/datastore")
+  end
+
+  describe "#completion_result" do
+    subject(:completion_result) { completion.completion_result }
+
+    let(:query) { "foo" }
+    let(:complete_query) { double("response", query_suggestions:) }
+    let(:query_suggestions) { %w[foo foobar foobaz].map { double("suggestion", suggestion: _1) } }
+
```

Comment:
I Imagine you want a test to assert that it returns a CompletionResult object?

Reply:
TBH I'm happy to just test behaviour rather than type here (which it does by calling `#suggestions` on the returned subject) but happy to add a test in for that if you think it'd be useful?

Reply:
Yeah seems ok, has same coverage. Don't worry about an extra test. 

This does feel a bit nuanced as this seems to just hold because CompletionRecord is super simple and not tested in it's own right.